From fa8fe5a7ffb1da6a4e252e10b74dfa683a087b36 Mon Sep 17 00:00:00 2001
From: Nishanth Peethambaran <nishanth.peethambaran@amd.com>
Date: Thu, 17 Mar 2016 03:17:11 -0400
Subject: [PATCH 1/3] st/omx/enc: Correct the timestamping

Attach the timestamp to the encoder task and use that timestamp
while pushing bitstrema buffer to the omx client.

Signed-off-by: Nishanth Peethambaran <nishanth.peethambaran@amd.com>
---
 src/gallium/state_trackers/omx/vid_enc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gallium/state_trackers/omx/vid_enc.c b/src/gallium/state_trackers/omx/vid_enc.c
index f427213..78d83bc 100644
--- a/src/gallium/state_trackers/omx/vid_enc.c
+++ b/src/gallium/state_trackers/omx/vid_enc.c
@@ -59,6 +59,7 @@ struct encode_task {
    struct list_head list;
 
    struct pipe_video_buffer *buf;
+   OMX_TICKS timestamp;
    unsigned pic_order_cnt;
    struct pipe_resource *bitstream;
    void *feedback;
@@ -1151,6 +1152,7 @@ static OMX_ERRORTYPE vid_enc_EncodeFrame(omx_base_PortType *port, OMX_BUFFERHEAD
       if (err != OMX_ErrorNone)
          return err;
    }
+   task->timestamp = buf->nTimeStamp;
 
    /* -------------- determine picture type --------- */
    if (!(priv->pic_order_cnt % OMX_VID_ENC_IDR_PERIOD_DEFAULT) ||
@@ -1249,4 +1251,5 @@ static void vid_enc_BufferEncoded(OMX_COMPONENTTYPE *comp, OMX_BUFFERHEADERTYPE*
 
    output->nOffset = 0;
    output->nFilledLen = size; /* mark buffer as full */
+   output->nTimeStamp = task->timestamp;
 }
-- 
2.1.4

